### ISSUE TITLE
Custom Builds

### ISSUE THREAD

**Issue by shshaw** [kenwheeler/cash#94]
A custom build process that would allow for excluding modules would be useful. Make micro-cash libraries with only the specific features your project needs.

**Comment by shshaw** [comment]
Using `preprocess`, you could implement it in this way in `src/_wrapper.js` by feeding an excludes object, probably from the command line parameters.

  // @ifndef excludes.attributes
  // @include ./attributes.js
  // @endif
  // @ifndef excludes.collection
  // @include ./collection.js
  // @endif
  // @ifndef excludes.css
  // @include ./css.js
  // @endif
  // @ifndef excludes.dimensions
  // @include ./dimensions.js
  // @endif
  // @ifndef excludes.events
  // @include ./events.js
  // @endif
  // @ifndef excludes.forms
  // @include ./forms.js
  // @endif
  // @ifndef excludes.manipulation
  // @include ./manipulation.js
  // @endif
  // @ifndef excludes.offset
  // @include ./offset.js
  // @endif
  // @ifndef excludes.traversal
  // @include ./traversal.js
  // @endif

**Comment by shshaw** [comment]
Another thought on this: Cash could be a great utility for currently jQuery-only plugins. If we do the custom build process granularly enough, some plugins could go jQuery-free by only adding 2-5kb worth of Cash without having to change their codebase much.

**Comment by simeydotme** [comment]
@shshaw , please refer to my comment in #92 :)

Also; thought... Have you ever used Modernizr's build tool? ( https://www.npmjs.com/package/gulp-modulizr ) for detecting what parts of modernizr your application needs, and then creating a custom build? I wonder how difficult that would be?

**Comment by shshaw** [comment]
Is that where you just check off the specific features you want? I do not like the [latest Modernizr's version of that](https://modernizr.com/download?setclasses) (way too hard to find the features you want), but the old one was easy to use. 

I do like that idea, as it's much more straightforward than having to clone the repo, install the NPM packages and create your own build, but I have no idea how difficult that would be to implement on a server.

**Comment by simeydotme** [comment]
no no, the gulp tool will actually go through your `.js` files, and find all calls for `modernizr` and add the features you need to the file. -- obviously it's a little different with `cash` as we'd have to use regex to find all the different methods on the prototype.

**Comment by shshaw** [comment]
As I mentioned above, `core.js` and `util.js` are essential to all the modules, and `data.js` is required for `events.js`. Other than that, every separate .js file should already be modular.

Yes, it should be a matter of just indexing the cash prototype and checking all `$()` calls. Not sure exactly how to implement that, but I would assume it's not terribly difficult.

**Comment by fabiospampinato** [comment]
Fixed in v2.1.2, the project is now bundles using [pacco](https://github.com/fabiospampinato/pacco).

Usually other modularized applications don't allow very granular partial builds, for instance they might let you skip the "traversal" module all together, but not just a single module.

The way I refactored the project it's now possible to exclude single files from the bundle, and even replace specific ones with custom files, in case somebody needs this.
